### PR TITLE
[5.0] Remove outdated captcha for new installation

### PR DESCRIPTION
**Pull request by Fedik**
Pull Request for Issue #39909 .

### Summary of Changes

Remove outdated captcha from core.
This still will keep the invisible captcha for existing installations, and uninstall old recaptcha.

### Testing Instructions

Do new installation.
Try update from 4.4.

### Actual result BEFORE applying this Pull Request
Old Cpatcha still there.

### Expected result AFTER applying this Pull Request
Do new installation: no captcha
Try update from 4.4: Invisible cpatcha still there,  and old recaptcha is uninstalled

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: 
- [ ] No documentation changes for docs.joomla.org needed
- [x] Pull Request link for manual.joomla.org: https://github.com/joomla/Manual/pull/182 Someone have to write a good explanation, and what to do for Users.
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: administrator/components/com_admin/sql/updates/mysql/5.0.0-2023-08-30.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+UPDATE `#__extensions`
```
Comment (wilsonge) [inline administrator/components/com_admin/sql/updates/mysql/5.0.0-2023-08-30.sql:1]: This is going to remove it on upgrades...

**Comment by Fedik** [comment]
> I really don't understand why we are removing invisible?

We can keep it. As you guys decide. For me no difference.

Invisible captcha is not deprecated but ia also outdated, there already recaptcha v3.

**Comment by brianteeman** [comment]
previously I had opened an issue for this https://github.com/joomla/joomla-cms/issues/39909 but I closed it yesterday after 6 months with no interest.

PS v3 is a different thing to v2 and invisible and not a direct replacement

**Comment by richard67** [comment]
@Fedik It does not need the update SQL scripts for unlocking the extensions. We have a method for that in script.php:

https://github.com/joomla/joomla-cms/blob/5.0-dev/administrator/components/com_admin/script.php#L214

You just have to add the extension to be uninstalled to this array:

https://github.com/joomla/joomla-cms/blob/5.0-dev/administrator/components/com_admin/script.php#L221-L234

That's all. The extension will be unlocked and uninstalled.

**Comment by rdeutz** [comment]
We discussed it in todays maintainers meeting and decided: We remove it for new installations, unlock it and deinstall the not working one.

**Comment by brianteeman** [comment]
if only the maintainers published reports then interested parties would have a clue wha\t was going on :(

**Comment by Fedik** [comment]
I have updated the PR, recaptcha is uninstalled, and invisible one is unlocked

**Comment by HLeithner** [comment]
> if only the maintainers published reports then interested parties would have a clue wha\t was going on :(

The issue get my attention because you closed it and got into my inbox. it's hard to review 800 open issues for a release. So it was simple luck that it triggered my attention.

File: administrator/components/com_admin/sql/updates/mysql/5.0.0-2023-08-30.sql [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+   SET `locked` = 0
+ WHERE `element` = 'recaptcha_invisible' AND `folder` = 'captcha';
```
Comment (richard67) [inline administrator/components/com_admin/sql/updates/mysql/5.0.0-2023-08-30.sql]:
```suggestion
 WHERE `type` = 'plugin' AND `element` = 'recaptcha_invisible' AND `folder` = 'captcha';
```
We should be precise when identifying extensions.

File: administrator/components/com_admin/sql/updates/postgresql/5.0.0-2023-08-30.sql [hunk 3]
```diff
@@ -0,0 +1,3 @@
+UPDATE "#__extensions"
+   SET "locked" = 0
+ WHERE "element" = 'recaptcha_invisible' AND "folder" = 'captcha';
```
Comment (richard67) [inline administrator/components/com_admin/sql/updates/postgresql/5.0.0-2023-08-30.sql]:
```suggestion
 WHERE "type" = 'plugin' AND "element" = 'recaptcha_invisible' AND "folder" = 'captcha';
```

**Comment by brianteeman** [comment]
What happens if a site is using recaptcha? Is it still uninstalled?

Should an update server not be added for the remaining captcha? Otherwise if there was a security issue with it we wouldn't be able to resolve it.

Should there be a conditional post installation message?

**Comment by Fedik** [comment]
> What happens if a site is using recaptcha? Is it still uninstalled?

Yes

> Should an update server not be added for the remaining captcha?

This captcha also should be removed/replaced. 
Someone have to write a good explanation, and what to do for Users.

> Should there be a conditional post installation message?

If someone write it, I try to add it to the PR.

**Comment by HLeithner** [comment]
We would need a "What's new in joomla 5.0" splash screen after upgrade like other software has but nobody wrote it yet.

Postinstallation messages are not the coolest thing tbh. but better then nothing

**Comment by Fedik** [comment]
> Fedik says both should be removed but I read that Robert says just one of them

It not conflicting. The invisible captcha will stay after update.

What I mean, it should be replaced by User to better captcha, because it is useles against spam.

**Comment by HLeithner** [comment]
thanks, could you please add this to the migration documentation, thanks.

**Comment by Fedik** [comment]
There it is https://github.com/joomla/Manual/pull/182

**Comment by uzielweb** [comment]
Ok, guys. but WHY I have no one captcha as default now????????????

**Pull request merged**